teamMaster.py

```diff
@@ -0,0 +1,264 @@
+Load daily schedule dictionary into memory
+Use schedule to search thru boxscore dictionaries to get all team info
+Update teamMaster dictionary with any new team data found
+"""
+
+
+import sys
+import shutil
+import argparse
+import logging
+import logging.config
+import datetime
+import json
+import requests
+
+
+# setup global variables
+SCRIPT = 'teamMaster.py'
+LOGGING_INI = 'teamMaster_logging.ini'
+DAILY_SCHEDULE = 'Data/schedule_YYYYMMDD.json'
+TEAM_MSTR_I = 'Data/teamMaster.json'
+TEAM_MSTR_O = 'Data/teamMaster_YYYYMMDD.json'
+BOXSCORE = 'http://gd2.mlb.com/_directory_/boxscore.json'
+
+
+class LoadDictionaryError(ValueError):
+    pass
+
+
+def init_logger():
+    """
+    initialize global variable logger and set script name
+    """
+
+    global logger
+    logger = logging.getLogger(SCRIPT)
+
+
+def get_command_arguments():
+    """
+    --date  optional, will extract teams found in boxscore for this date
+    """
+
+    parser = argparse.ArgumentParser('teamMaster command line arguments')
+    parser.add_argument(
+        '-d',
+        '--date',
+        help='Date of daily schedule for input - mm-dd-yyyy format',
+        dest='game_date',
+        type=str)
+
+    argue = parser.parse_args()
+    logger.info('Command arguments: ' + str(argue.game_date))
+
+    return argue
+
+
+def determine_filenames(gamedate=None):
+    """
+    :param game_date: date of the games in format "MM-DD-YYYY"
+    """
+
+    if gamedate is not None:
+        yyyy = gamedate[6:10]
+        mm = gamedate[0:2]
+        dd = gamedate[3:5]
+    else:
+        yyyy = datetime.date.today().strftime("%Y")
+        mm = datetime.date.today().strftime("%m")
+        dd = datetime.date.today().strftime("%d")
+
+    mmddyyyy = mm + '-' + dd + '-' + yyyy
+    schedule_input = DAILY_SCHEDULE.replace('YYYYMMDD', yyyy + mm + dd)
+    team_output = TEAM_MSTR_O.replace('YYYYMMDD', yyyy + mm + dd)
+
+    logger.info('dailySched dictionary location: ' + schedule_input)
+    logger.info('Updated team dictionary location: ' + team_output)
+
+    return (schedule_input, team_output, mmddyyyy)
+
+
+def load_daily_schedule(schedule_in):
+    """
+    :param schedule_in: filename for daily schedule file to load into memory
+    """
+
+    logger.info('Loading dailySchedule dictionary')
+
+    try:
+        with open(schedule_in) as schedulefile:
+            return json.load(schedulefile)
+    except Exception as e:
+        errmsg = 'Error loading dailySchedule dictionary. . .'
+        logger.critical(errmsg)
+        logger.exception(e)
+        raise LoadDictionaryError(errmsg)
+
+
+def reset_todays_games_in_master(masterdict):
+
+    keylist = list(masterdict)
+    for key in keylist:
+        keys = list(masterdict[key])
+        if 'today_1' in keys:
+            del masterdict[key]['today_1']
+        if 'today_2' in keys:
+            del masterdict[key]['today_2']
+
+    return masterdict
+
+
+def process_schedule(sched_dict, mstr_dict):
```

Comment:
Still looking at this, think I have an idea to make it smaller. Because of the lengthy amount of logic within the large IF statement that is already broken into smaller functions, it's not as easy to make is smaller, but will try.